dask_sql/physical/rel/custom/create_model.py

```diff
@@ -156,20 +171,21 @@ def convert(
             except ImportError:  # pragma: no cover
                 raise ValueError("Wrapping requires dask-ml to be installed.")
 
-            model = ParallelPostFit(estimator=model)
+            # When `wrap_predict` is set to True we train on single partition frames
+            # because this is only useful for non dask distributed models
+            # Training via delayed fit ensures that we dont have to transfer
+            # data back to the client for training
 
-        select_query = context._to_sql_string(select)
-        training_df = context.sql(select_query)
+            X_d = X.repartition(npartitions=1).to_delayed()
+            if y is not None:
+                y_d = y.repartition(npartitions=1).to_delayed()
+            else:
+                y_d = None
```

Comment:
For more context around this see issue: https://github.com/rapidsai/cuml/issues/4406 .

We were previously training on client which is:

a. Very inefficient and possibly problematic in multi-node clusters and heterogeneous setup. 
b. Training non distributed xgboost models on dask collections is not supported.